File: spark/src/test/scala/geotrellis/spark/io/SpatialKeyIndexMethods.scala
package geotrellis.spark.io

import geotrellis.spark._
import geotrellis.spark.io.index._

trait SpatialKeyIndexMethods {
  def keyIndexMethods: Map[String, KeyIndexMethod[SpatialKey]] =
    Map(
      "row major" -> RowMajorKeyIndexMethod,
      "z order" -> ZCurveKeyIndexMethod,
      "hilbert" -> HilbertKeyIndexMethod
    )
}

File: geotrellis-python/tests/geotrellis/spark/io/SpatialKeyIndexMethods.py
from geotrellis.spark.io.index.RowMajorKeyIndexMethod import RowMajorKeyIndexMethod
from geotrellis.spark.io.index.ZCurveKeyIndexMethod import ZCurveKeyIndexMethod
from geotrellis.spark.io.index.HilbertKeyIndexMethod import HilbertKeyIndexMethod

class _SpatialKeyIndexMethods(object):
    @property
    def keyIndexMethods(self):
        return {
                "row major": RowMajorKeyIndexMethod,
                "z order": ZCurveKeyIndexMethod,
                "hilbert": HilbertKeyIndexMethod}
